src/sql/statement.rs

```diff
@@ -1339,6 +1371,22 @@ fn handle_create_source(scx: &StatementContext, stmt: Statement) -> Result<Plan,
                         Some(Value::SingleQuotedString(topic)) => Consistency::BringYourOwn(topic),
                         Some(_) => bail!("consistency must be a string"),
                     };
+
+                    max_ts_batch = match with_options.remove("max_ts_batch") {
+                        None => 0,
+                        Some(Value::Number(n)) => match n.parse::<i64>() {
+                            Ok(n) => {
+                                if n < 0 {
+                                    bail!("max_ts_batch must be greater than zero")
+                                } else {
+                                    n
+                                }
+                            }
+                            _ => bail!("max_ts_batch must be an i64"),
+                        },
+                        Some(_) => bail!("max_ts_batch must be an i64"),
+                    };
```

Comment:
Can you extract a helper function to do the with option extraction and parsing?

Reply:
Also, this shouldn't really be an i64, right? I'd say we should parse it as a usize here.

Reply:
It is an i64 because (confusingly) offsets are i64s in Kafka, so I wanted to make sure the user did not try to specify a batch size that is greater than an i64. Happy to move this to usize and to move the conversion inside timestamp.rs